### ISSUE TITLE
Login Link Email: Prevent plain text turning into links in some email clients

### ISSUE THREAD

**Issue by ivan-ottinger** [Automattic/wp-calypso#74194]
In some email clients, "WordPress.com" gets turned into a link automatically - we can try to prevent that behavior:

![Image](https://user-images.githubusercontent.com/25105483/223670527-1b16ce80-f51c-4958-8532-6af74b97c04d.png)

**Comment by yansern** [comment]
Have a feeling it won't be easy to trick Apple Mail because we've been down this road before.

**Comment by ivan-ottinger** [comment]
> Have a feeling it won't be easy to trick Apple Mail because we've been down this road before.

🤔 True. I remember we were looking into similar issue before, but don't remember the actual outcomes of the investigation.

**Comment by yansern** [comment]
> but don't remember the actual outcomes of the investigation.

Same here. Could revisit this issue again.

I'm foreseeing that the WordPress.com text remains intact in other translated languages.

We wouldn't want to add the modifications to the WordPress.com text directly pre-translation, but rather, fix it post-translation by doing a search & replace, and replacing it with some, html tags or unicode char trickery.

**Comment by yansern** [comment]
Found a possible solution related to `x-apple-data-detectors`. Context here: p1685665923040259/1685641660.502559-slack-C02TCEHP3HA

**Comment by yansern** [comment]
Need to identify which emails are affected and where and then split them into smaller clearer tasks.

**Comment by TimBroddin** [comment]
@yansern I've read your analysis, and it's a very clean solution. However, how would this work when we inline all CSS?

**Comment by TimBroddin** [comment]
Loose idea: insert a non-visible space to prevent data detection. But this will break stuff if the user tries to copy/paste.

**Comment by TimBroddin** [comment]
Other idea: ship inline `<style></style>` containing the overrides for Apple Mail. However, I think the emogrifier will eat all our applied classes.

**Comment by TimBroddin** [comment]
I did a bunch of testing and here are my conclusions:

### 1. No proof of x-apple-data-detectors attribute
I don't think `x-apple-data-detectors` attributes get added to the links. This is how it happens in mobile Safari, but I couldn't verify it in Apple Mail.  Both injecting CSS targeting `a[x-apple-data-detectors]`and inserting the meta tag Apple suggests for Safari didn't have any effect:

```<meta name="format-detection" content="telephone=no,date=no,address=no,email=no,url=no" />```

It's hard to debug the rendering engine of Safari, but I see no evidence of this attribute getting added.

### 2. Targeting these links with a selector (works)
I was able to inject this snippet, and get styling to work by wrapping the string in a `<span class="apple-detector-fox"></span> `

```css
 <style>
        .apple-detector-fix a {
            color: inherit;
            text-decoration: inherit;
        }
</style>
```

### 3. Zero-width spaces
I was also capable of defeating the link detection by inserting some [zero width spaces](https://symbl.cc/en/200B/) inside the URL. We could write a filter that takes all links inside `<span class="no-links"></span>` (for example) and inserts zero-width spaces inside them. The upside (as opposed to point 2) is that the links are no longer clickable. The downside is that I assume this might cause an increase in false positive spam detection, since zero width spaces in URL's seem very fishy.

**Comment by TimBroddin** [comment]
Just found this inside `template-styles.css`, so someone must have once faced the same issue:

```css
	.apple-link a {
		color: inherit !important;
		font-family: inherit !important;
		font-size: inherit !important;
		font-weight: inherit !important;
		line-height: inherit !important;
		text-decoration: none !important;
	}
```

So what we'd need to do is wrap all links we don't want linkified with `<span class="apple-link"></span>`.

**Comment by TimBroddin** [comment]
Fix in: D114820-code
